core/src/server/rpc_module.rs

```diff
@@ -221,6 +322,101 @@ impl Methods {
 		Self::default()
 	}
 
+	/// Helper for obtaining a mut ref to the callbacks HashMap.
+	///
+	/// ## Warning
+	///
+	/// Direct access to mut callbacks is discouraged and can result in
+	/// dangling handlers (e.g. a [`MethodCallback::Subscription`] with no
+	/// corresponding [`MethodCallback::Unsubscription`]). As such this method
+	/// should be used with caution.
+	///
+	/// To access the underlying map correctly, you MUST NOT perform the
+	/// following operations without checking for dangling handlers:
+	///
+	/// - `mut_callbacks().remove()`
+	/// - `mut_callbacks().insert()`
+	/// - `mut_callbacks().entry()`
+	/// - `mut_callbacks().drain()`
+	/// - `mut_callbacks().iter_mut()`
+	///
+	/// Prefer using the provided methods:
+	/// - [`Self::remove`]
+	/// - [`Self::replace`]
+	/// - [`Self::verify_and_insert`]
+	/// - [`Self::replace_if`]
+	/// - [`Self::merge`]
+	/// - [`Self::merge_replace`]
+	/// - [`Self::merge_replace_if`]
+	fn mut_callbacks(&mut self) -> &mut FxHashMap<&'static str, MethodCallback> {
+		Arc::make_mut(&mut self.callbacks)
+	}
+
+	/// Returns the method callback.
+	pub fn method(&self, method_name: &str) -> Option<MethodCallback> {
+		match self.callbacks.get(method_name) {
+			Some(MethodCallback::Alias(alias)) => alias.upgrade(),
+			other => other.cloned(),
+		}
+	}
+
+	/// Returns an `Iterator` with all the method names registered on this server.
+	pub fn method_names(&self) -> impl Iterator<Item = &'static str> + '_ {
+		self.callbacks.keys().copied()
+	}
+
+	/// Returns the method callback along with its name. The returned name is same as the
+	/// `method_name`, but its lifetime bound is `'static`.
+	pub fn method_with_name(&self, method_name: &str) -> Option<(&'static str, MethodCallback)> {
+		self.callbacks.get_key_value(method_name).and_then(|(k, v)| {
+			let v = match v {
+				MethodCallback::Alias(alias) => alias.upgrade()?,
+				other => other.clone(),
+			};
+			Some((*k, v))
+		})
+	}
+
+	/// Register a new alias for a target method, or returns an error if the
+	/// alias name was already taken. If the target method does not exist, the
+	/// alias will not be registered.
+	pub fn register_alias(
+		&mut self,
+		alias: &'static str,
+		target_name: &'static str,
+	) -> Result<(), RegisterMethodError> {
```

Comment:
We returned an error if the target_name wasn't not registered before and I would like to keep it that way.

Why did you change that (maybe I'm missing something)?